### PR TITLE
Add listAssist to RegionalProcessingCenter and add hearing to caseData

### PR DESCRIPTION
**Pull request by MohammedHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10222

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java [hunk 1]
```diff
@@ -21,7 +21,8 @@ public RegionalProcessingCenter(@JsonProperty("faxNumber") String faxNumber,
                                     @JsonProperty("address3") String address3,
                                     @JsonProperty("postcode") String postcode,
                                     @JsonProperty("city") String city,
-                                    @JsonProperty("email") String email) {
+                                    @JsonProperty("email") String email,
+                                    @JsonProperty("listAssist") String listAssist){
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java:25]: Is this the field that Jonathan was working on? If so I think this ought to be a boolean

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java [hunk 2]
```diff
@@ -219,6 +219,8 @@
     private String adjournCaseNextHearingDateOrTime;
     private String adjournCaseNextHearingFirstAvailableDateAfterDate;
     private String adjournCaseNextHearingFirstAvailableDateAfterPeriod;
+    private String hearingRoute;
+    private StateOfHearing hearingState;
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java:223]: The field name should be Hearing State

File: src/main/resources/reference-data/rpc-data.json [hunk 3]
```diff
@@ -9,7 +9,8 @@
     "postcode" : "L2 5UZ",
     "phoneNumber" : "0300 123 1142",
     "faxNumber" : "0870 324 0109",
-    "email" : "Liverpool_SYA_Resp@justice.gov.uk"
+    "email" : "Liverpool_SYA_Resp@justice.gov.uk",
+    "listAssist" : "false"
```
Comment (dattipoeHMCTS) [inline src/main/resources/reference-data/rpc-data.json:13]: JSON can handle boolean values - no need for the quotes

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java [hunk 4]
```diff
@@ -219,6 +219,8 @@
     private String adjournCaseNextHearingDateOrTime;
     private String adjournCaseNextHearingFirstAvailableDateAfterDate;
     private String adjournCaseNextHearingFirstAvailableDateAfterPeriod;
+    private String hearingRoute;
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java:222]: These two should be on the Hearings object, HearingDetails I believe. I'd also make the field an enum so we have a solid definition of what each type is.

I'd also probably call it something like HearingBookingType to make it clearer.

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java [hunk 5]
```diff
@@ -32,25 +33,28 @@ public RegionalProcessingCenter(@JsonProperty("faxNumber") String faxNumber,
         this.postcode = postcode;
         this.city = city;
         this.email = email;
+        this.hearingRoute = hearingRoute;
     }
 
-    private String faxNumber;
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java:37]: Why'd you remove all the access modifiers?

**Comment by dattipoeHMCTS** [comment]
Branch name as well

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java (hunk 5 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java:37]: We are using the lombok @Value which set the access profile to private anyway so seemed a bit redundant to have it twice

**Pull request closed without merging**